arch/arm64/core/cortex_r/arm_mpu.c

```diff
@@ -183,4 +198,293 @@ void z_arm64_mm_init(bool is_primary_core)
 	static_regions_num = mpu_config.num_regions;
 
 	arm_core_mpu_enable();
+
+#ifdef CONFIG_USERSPACE
+	int rc = dynamic_areas_init(_MPU_DYNAMIC_REGIONS_AREA_START,
+				    _MPU_DYNAMIC_REGIONS_AREA_SIZE);
+	if (rc <= 0) {
+		__ASSERT(0, "Dynamic areas init fail");
+		return;
+	}
+#endif
+
+}
+
+#ifdef CONFIG_USERSPACE
+
+struct dynamic_region_info {
+	int index;
+	struct arm_mpu_region region_conf;
+};
+
+static struct dynamic_region_info sys_dyn_regions[MPU_DYNAMIC_REGION_AREAS_NUM];
+static int sys_dyn_regions_num;
+
+static int dynamic_areas_init(uintptr_t start, size_t size)
+{
+	if (sys_dyn_regions_num + 1 > MPU_DYNAMIC_REGION_AREAS_NUM) {
+		return -1;
+	}
+	uint64_t base = start;
+	uint64_t limit = base + size;
+	int i;
+
+	const struct arm_mpu_region *region;
+	struct dynamic_region_info *tmp_info;
```

Comment:
nit: can you move this at the beginning of the function?

Reply:
Sure, I will